derp/derp_server.go

```diff
@@ -215,6 +224,24 @@ func (s *Server) PrivateKey() key.Private { return s.privateKey }
 // PublicKey returns the server's public key.
 func (s *Server) PublicKey() key.Public { return s.publicKey }
 
+// emptyBytes returns a (possibly-reused) byte slice of the given length.
+func emptyBytes(length int) []byte {
+	zeroed := bytesPool.Get().(*[]byte)
```

Comment:
why `zeroed`? It's not necessarily.

I'd write:

 ```go
      bp, _ := bytesPool.Get().(*[]byte)
```

Note that using the two-lvalue "comma-ok" form on the LHS, even when throwing away the `ok bool` part into `_` guarantees that the interface type assertion to `*[]byte` never fails.